### PR TITLE
Add simulator targets to several FPGA Tutorials

### PR DESCRIPTION
**Pull request by ajaykumarkannan**
# Existing Sample Changes
## Description

Add simulator targets for the following designs: 

- ac_fixed
- ac_int
- autorun
- dsp_control
- fpga_reg
- latency_control
- scheduler_target_fmax

Fixes Issue# 

## External Dependencies

List any external dependencies created as a result of this change.

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Implement fixes for ONSAM Jiras

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [ ] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
 # Autorun Kernels
+
 This FPGA tutorial demonstrates how to create the equivalent of OpenCL `autorun` kernels in oneAPI. An autorun kernel is one that is submitted before the `main()` function begins, and typically (though not necessarily) never finishes.
 
 | Optimized for                     | Description
 ---                                 |---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md:7]: Was this change intended?

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md [hunk 2]
```diff
@@ -71,107 +78,152 @@ After learning how to use the extensions for Intel oneAPI Toolkits, return to th
...
      make report
      ```
-   * Compile for FPGA hardware (longer compile time, targets FPGA device):
+
+   - Compile for FPGA hardware (longer compile time, targets FPGA device):
+
+   - Compile for simulation (fast compile time, targets simulated FPGA device)
+
+     ```bash
+     make fpga_sim
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md:124]: This is out of order

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md [hunk 3]
```diff
@@ -71,107 +78,152 @@ After learning how to use the extensions for Intel oneAPI Toolkits, return to th
...
+
      ```
      ./autorun.fpga_emu     (Linux)
      set SYCL_ENABLE_DEFAULT_CONTEXTS=1 && autorun.fpga_emu.exe   (Windows)
      ```
-Note that when running this particular sample in emulation on Windows, an environment variable must be set as shown or the execution will hang.
 
+Note that when running this particular sample in emulation on Windows, an environment variable must be set as shown or the execution will hang.
 
 2. Run the sample on the FPGA device:
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md:202]:
```suggestion
2. Run the sample on the FPGA simulator device:
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ This FPGA tutorial demonstrates how to use the Algorithmic C (AC) data type `ac_
 
 | Optimized for                     | Description
 |:---                               |:---
-| OS                                | CentOS* Linux 8 <br> Red Hat* Enterprise Linux* 8 <br> SUSE* Linux Enterprise Server 15 <br> Ubuntu* 18.04 LTS <br> Ubuntu 20.04 <br>Windows* 10
+| OS                                | CentOS*Linux 8 <br> Red Hat* Enterprise Linux*8 <br> SUSE* Linux Enterprise Server 15 <br> Ubuntu*18.04 LTS <br> Ubuntu 20.04 <br>Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/README.md:7]: Is this intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/README.md [hunk 5]
```diff
@@ -3,8 +3,8 @@
 This FPGA tutorial demonstrates how to use the Algorithmic C (AC) data type `ac_int` and some best practices.
 
 | Optimized for                     | Description
-|:---                                 |:---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+|:---                               |:---
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/README.md:7]: Intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/README.md [hunk 6]
```diff
@@ -1,18 +1,21 @@
 # DSP Control
+
 This FPGA tutorial demonstrates how to set the implementation preference for certain math operations (addition, subtraction, and multiplication by a constant) between hardened DSP blocks and soft logic.
 
 | Optimized for                     | Description
 |:---                                 |:---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/README.md:7]: Intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/README.md [hunk 7]
```diff
@@ -1,19 +1,22 @@
 # Latency Control
+
 This FPGA tutorial demonstrates how to set latency constraints to pipes and load-store units (LSUs) accesses.
 
 | Optimized for                     | Description
 |:---                                 |:---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/README.md:7]: Intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/README.md [hunk 8]
```diff
@@ -233,36 +283,49 @@ Intel&reg;oneAPI Toolkits, which provides system checks to find missing
...
 ## Running the Sample
 
  1. Run the sample on the FPGA emulator (the kernel executes on the CPU):
+
      ```
      ./latency_control.fpga_emu     (Linux)
      latency_control.fpga_emu.exe   (Windows)
      ```
+
 2. Run the sample on the FPGA device:
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/README.md:307]:
```suggestion
2. Run the sample on the FPGA simulator device:
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/README.md [hunk 9]
```diff
@@ -6,8 +6,8 @@ This FPGA tutorial demonstrates how a power user can apply the SYCL*-compliant C
 
 | Optimized for                     | Description
 ---                                 |---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
-| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA <br> Intel® FPGA 3rd party / custom platforms with oneAPI support <br> *__Note__: Intel® FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/README.md:9]: Intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md [hunk 10]
```diff
@@ -1,58 +1,66 @@
-
 # Avoiding Aliasing of Kernel Arguments
+
 This tutorial explains the  `kernel_args_restrict` attribute and its effect on the performance of FPGA kernels.
 
 | Optimized for                     | Description
 |:---                               |:---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md:7]: Intended?

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md [hunk 11]
```diff
@@ -61,105 +69,145 @@ In this tutorial, we will show how to use the `kernel_args_restrict` attribute f
...
 
-   * Compile for emulation (fast compile time, targets emulated FPGA device):
+   - Compile for emulation (fast compile time, targets emulated FPGA device):
+
       ```
       make fpga_emu
       ```
-   * Generate the optimization report:
+
+   - Compile for simulation (fast compile time, targets simulated FPGA device)
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md:142]: Just for consistency, let's have emulation -> report -> simulation -> hw

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md [hunk 12]
```diff
@@ -61,105 +69,145 @@ In this tutorial, we will show how to use the `kernel_args_restrict` attribute f
...
 
-   * Compile for emulation (fast compile time, targets emulated FPGA device):
+   - Compile for emulation (fast compile time, targets emulated FPGA device):
+
      ```
      nmake fpga_emu
      ```
-   * Generate the optimization report:
+
+   - Compile for the simulation target (fast compile time, targets simulated FPGA device)
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md:197]: Same comment about the order

File: DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/README.md [hunk 13]
```diff
@@ -1,32 +1,38 @@
 # Scheduler Target FMAX
+
 This tutorial explains the `scheduler_target_fmax_mhz` attribute and its effect on the performance of Intel&reg; FPGA kernels.
 
 | Optimized for                     | Description
 |:---                               |:---
-| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| OS                                | Linux*Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE*15 <br> Windows* 10
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/README.md:7]: Intended?

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Comment by yuguen-intel** [comment]
Please use the `target` instead of `board` in the newly added cmake commands to transition nicely with: https://github.com/oneapi-src/oneAPI-samples/pull/1111/files

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md (hunk 1 above)
Comment (ajaykumarkannan) [reply DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/README.md:7]: Nope, typo.

**Comment by ajaykumarkannan** [comment]
> Please use the `target` instead of `board` in the newly added cmake commands to transition nicely with #1111

I merged the changes from there - will submit #1111 first before this one.

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/README.md [hunk 14]
```diff
@@ -79,6 +79,11 @@ The following instructions assume you are in the project's root folder.
      make report
      ```
      Locate the FPGA optimization report, `report.html`, in the `fpga_report.prj/reports/` directory.
+   - Use the following commands to build and run the template for FPGA simulator:
+     ```
+     make fpga_sim
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/README.md:84]: Wow I did not even know about this sample.
We don't even test it :/
Paul is working on a similar but up to date sample, I'll ask for it to be replaced + moved to the FPGA part of the code samples.

**Review by yuguen-intel — APPROVED** [review]

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/decompress/src/CMakeLists.txt [hunk 15]
```diff
@@ -33,8 +33,20 @@ else()
...
-set(AC_TYPES_FLAG "-qactypes")
+# These are Windows-specific flags:
+# 1. /EHsc This is a Windows-specific flag that enables exception handling in host code
+# 2. /Qactypes Include ac_types headers and link against ac_types emulation libraries
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+    set(AC_TYPES_FLAG "/Qactypes")
+    # Increase the allowable constexpr steps for the front end. This allows the
+    # front-end compiler to do more compile-time computation.
+    set(CONSTEXPR_STEPS "/constexpr:steps5084968")
```
Comment (ajaykumarkannan) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/decompress/src/CMakeLists.txt:44]: Is there a typo on this line?

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/src/CMakeLists.txt [hunk 16]
```diff
@@ -1,6 +1,7 @@
 set(SOURCE_FILE main.cpp)
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/src/CMakeLists.txt:1]: I'm not looking into this file too much as we are going to remove this sample entirely for this release.
After you pointed me to it, I looked at it and realized it was:
1/ outdated
2/ in a place where FPGA users would not find it

Paul will be adding a replacement to this sample in DPC++FPGA/Tutorial/GettingStarted

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/src/CMakeLists.txt [hunk 17]
```diff
@@ -62,9 +65,23 @@ target_include_directories(${FPGA_EARLY_IMAGE} PRIVATE ../../../../include)
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR ${AC_TYPES_FLAG} -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> ac_fixed.cpp -o ac_fixed.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR ${AC_TYPES_FLAG} -o ac_fixed.cpp.o -c ac_fixed.cpp
+#   [link]    icpx -fsycl -fintelfpga ${AC_TYPES_FLAG} -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> ac_fixed.cpp.o -o ac_fixed.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_fixed/src/CMakeLists.txt:78]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/src/CMakeLists.txt [hunk 18]
```diff
@@ -59,6 +62,21 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> autorun.cpp -o autorun.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -o autorun.cpp.o -c autorun.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> autorun.cpp.o -o autorun.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/src/CMakeLists.txt:73]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/CMakeLists.txt [hunk 19]
```diff
@@ -66,6 +70,20 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#    icpx -fsycl -fintelfpga ${AC_TYPES_FLAG} -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> fpga_compile.cpp -o fpga_compile.fpga_emu
+# CMake executes:
+#    [compile] icpx -fsycl -fintelfpga ${AC_TYPES_FLAG} -DFPGA_SIMULATOR -o fpga_compile.cpp.o -c fpga_compile.cpp
+#    [link]    icpx -fsycl -fintelfpga ${AC_TYPES_FLAG} -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> fpga_compile.cpp.o -o fpga_compile.fpga_emu
+add_executable(${SIMULATOR_TARGET} ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/src/CMakeLists.txt:81]:
```suggestion
add_executable(${SIMULATOR_TARGET} ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/src/CMakeLists.txt [hunk 20]
```diff
@@ -59,6 +62,21 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> -Xsdsp-mode=prefer-softlogic dsp_control.cpp -o dsp_control.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -o dsp_control.cpp.o -c dsp_control.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> -Xsdsp-mode=prefer-softlogic dsp_control.cpp.o -o dsp_control.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/src/CMakeLists.txt:73]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/src/CMakeLists.txt [hunk 21]
```diff
@@ -68,6 +70,19 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> latency_control.cpp -o latency_control.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -o latency_control.cpp.o -c latency_control.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> latency_control.cpp.o -o latency_control.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/latency_control/src/CMakeLists.txt:81]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/src/CMakeLists.txt [hunk 22]
```diff
@@ -67,6 +71,24 @@ set_target_properties(${FPGA_EARLY_IMAGE_REG} PROPERTIES COMPILE_FLAGS "${HARDWA
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> fpga_reg.cpp -o fpga_reg.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -o fpga_reg.cpp.o -c fpga_reg.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> fpga_reg.cpp.o -o fpga_reg.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/src/CMakeLists.txt:82]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/src/CMakeLists.txt [hunk 23]
```diff
@@ -67,6 +71,24 @@ set_target_properties(${FPGA_EARLY_IMAGE_REG} PROPERTIES COMPILE_FLAGS "${HARDWA
...
+add_executable(${SIMULATOR_TARGET_REG} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/fpga_reg/src/CMakeLists.txt:83]:
```suggestion
add_executable(${SIMULATOR_TARGET_REG} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET_REG} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/src/CMakeLists.txt [hunk 24]
```diff
@@ -59,6 +62,21 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> kernel_args_restrict.cpp -o kernel_args_restrict.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -DFPGA_SIMULATOR -o kernel_args_restrict.cpp.o -c kernel_args_restrict.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> kernel_args_restrict.cpp.o -o kernel_args_restrict.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/src/CMakeLists.txt:73]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

File: DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/src/CMakeLists.txt [hunk 25]
```diff
@@ -59,6 +62,21 @@ set_target_properties(${FPGA_EARLY_IMAGE} PROPERTIES COMPILE_FLAGS "${HARDWARE_C
...
 
+###############################################################################
+### FPGA Simulator
+###############################################################################
+# To compile in a single command:
+#   icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> scheduler_target_fmax.cpp -o scheduler_target_fmax.fpga
+# CMake executes:
+#   [compile] icpx -fsycl -fintelfpga -o scheduler_target_fmax.cpp.o -c scheduler_target_fmax.cpp
+#   [link]    icpx -fsycl -fintelfpga -Xssimulation -Xsghdl -Xstarget=<FPGA_DEVICE> scheduler_target_fmax.cpp.o -o scheduler_target_fmax.fpga
+add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/src/CMakeLists.txt:73]:
```suggestion
add_executable(${SIMULATOR_TARGET} EXCLUDE_FROM_ALL ${SOURCE_FILE})
target_include_directories(${SIMULATOR_TARGET} PRIVATE ../../../../include)
```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**